src/Block/NonceFieldBlock.php

```diff
@@ -36,7 +41,18 @@ public function __construct(
     ) {
         $this->_setNonce($nonce)
             ->_setFieldName($fieldName)
-            ->_setUseReferer($useReferer);
+            ->_setUseReferer($useReferer)
+            ->_setRefererFieldName(static::REFERER_FIELD_NAME);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _getRefererUrl()
+    {
+        return \esc_attr(\wp_unslash($_SERVER['REQUEST_URI']));
```

Comment:
`esc_attr()` should only be used where you are using it as an attribute. This method does not need to know about how it's being used. Also, if you wanna do proper WP VIP,  you should at least use [`filter_input()`] with `$filter` left default. Ideally, of course, you would use something like [PSR-7], but at this point it feels overly complicated and unfeasible.

[`filter_input()`]:              http://php.net/manual/en/function.filter-input.php
[PSR-7]:                          http://www.php-fig.org/psr/psr-7/